addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -65,12 +65,35 @@ export class MessageList extends Component {
         // fields accessed during useRenderedValues should be observed by
         // useModels as if they were part of the OWL rendering itself.
         useModels();
+        /**
+         * This observer will trigger each time the root element is resized and
+         * adjust the scroll position.
+         */
+        this.resizeObserver = new ResizeObserver(elements => {
+            // In this case, this loop seems useless since we are really working
+            // on a single element, but it's required by the ResizeObserver to
+            // avoid error: ResizeObserver loop completed with undelivered
+            // notifications.
+            for (const element of elements) {
+                if (this.threadView && this.threadView.isAtEnd) {
+                    this._adjustScrollForExtraMessagesAtTheEnd();
+                }
+            }
+        });
         // useUpdate must be defined after useRenderedValues, indeed they both
         // use onMounted/onPatched, and the calls from useRenderedValues must
         // happen first to save the values before useUpdate accesses them.
         useUpdate({ func: () => this._update() });
     }
 
+    mounted() {
+        this.resizeObserver.observe(this._getScrollableElement());
+    }
```

Comment:
`_getScrollableElement` being a function, its return value might change, which would cause issues here (not properly observing again).

We should probably duplicate the logic once here with this.el and once in the chatter. It should be getting easier to do now that most of the logic has been moved into models.

Reply:
I don't understand this comment.
`_getScrollableElement`is `this.el` or a props. All of this code should belong to the component since it's related to the message list itself.
I don't understand your point on the model here. Should I try to use the model for the resize 

Both chat window and chatter use the message_list, I don't see how or why we should duplicate that inside the chatter (beside the fact that this has no real usefulness inside the chatter yet).

Reply:
I want to get rid of `_getScrollableElement` everywhere, and the only usage for it is on chatter because it has a different scrollbar.

With your new better code, it's possible for chatter to simply write the scroll values in its `threadView` directly, just like message list would write it here in the other cases (not chatter).